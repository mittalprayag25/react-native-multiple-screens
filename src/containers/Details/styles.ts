import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  header: {
    backgroundColor: '#ff6600',
    height: 50,
    justifyContent: 'center',
    paddingLeft: 10,
  },
  headerLabel: {
    fontWeight: 'bold',
    fontSize: 20,
    color: 'black',
  },
  subSection: {
    color: 'black',
  },
  comments: {
    flexDirection: 'column',
    justifyContent: 'space-around',
  },
});

export default styles;
