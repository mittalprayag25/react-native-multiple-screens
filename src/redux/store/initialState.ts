const initialState = {
  topFeed: {
    data: [],
    fullData: [],
    isFetching: false,
    error: null,
  },
  kids: {
    data: {},
    isFetching: false,
    error: null,
  },
};

export default initialState;
