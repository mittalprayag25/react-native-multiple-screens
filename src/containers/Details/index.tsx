import { Text, View, ScrollView } from 'react-native';
import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../redux/reducers';
import LoadingIndicator from './../../components/LoadingIndicator';
import CommentSection from './../../components/CommentSection';
import styles from './styles';

const Details = ({ navigation }: any): JSX.Element => {
  const feed = navigation.getParam('feed');
  const { data, isFetching }: any = useSelector((s: RootState) => s.kids);

  return (
    <ScrollView>
      <View style={styles.header}>
        <Text style={styles.headerLabel}>Hacker News</Text>
        <Text style={styles.subSection}>
          new | Past | comments | ask | show | jobs
        </Text>
      </View>

      <View>
        <Text style={styles.headerLabel}>{feed.title}</Text>
        <Text>Comments</Text>
        {isFetching ? <LoadingIndicator size={10} /> : null}
        {feed.text && <Text testID="detail-feed-text">{feed.text}</Text>}
        <View style={styles.comments}>
          {feed.kids && <CommentSection comments={feed.kids} data={data} />}
        </View>
      </View>
    </ScrollView>
  );
};

export default Details;
