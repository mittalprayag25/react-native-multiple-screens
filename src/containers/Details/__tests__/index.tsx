import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Linking} from 'react-native';
import * as reactRedux from 'react-redux';
import {Provider} from 'react-redux';

import {
  createMockStore,
  mockNavigationWithParams,
} from '../../../__mocks__/index';
import Details from '../index';
import initialState from './../../../redux/store/initialState';

const mockedDispatch = jest.fn();

describe('Details with success data', () => {
  const navigation = mockNavigationWithParams({
    feed: {
      by: 'timsneath',
      descendants: 99,
      id: 30827210,
      kids: [
        30828021, 30827448, 30827932, 30828013, 30827621, 30827406, 30827414,
        30827390, 30827521, 30827440, 30827379, 30827508, 30827335, 30827377,
        30827636, 30827711,
      ],
      time: 1648442500,
      text: 'Fuchsia Workstation',
      type: 'comment',
    },
  });

  const useSelectorSpy = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchSpy = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useDispatchSpy.mockReturnValue(mockedDispatch);
    useSelectorSpy.mockReturnValue({
      data: {
        30821604: {
          by: 'cute_boi',
          id: 30821604,
          kids: [30821923],
          parent: 30820894,
          text: 'comment text',
          time: 1648398990,
          type: 'comment',
        },
      },

      error: null,
      isFetching: false,
    });
    jest.setTimeout(10000);
  });

  const props = {
    navigation: navigation,
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(
      <Provider store={createMockStore(changeState)}>
        <Details {...props} />
      </Provider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('Should contain CLick More Text', () => {
    const {getByTestId} = render(
      <Provider store={createMockStore(changeState)}>
        <Details {...props} />
      </Provider>,
    );
    expect(getByTestId('detail-feed-text').props.children).toBe(
      'Fuchsia Workstation',
    );
  });
});

describe('Details with isFetching data', () => {
  const navigation = mockNavigationWithParams({
    feed: {
      by: 'timsneath',
      descendants: 99,
      id: 30827210,
      time: 1648442500,
      type: 'comment',
    },
  });

  const useSelectorSpy = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchSpy = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useDispatchSpy.mockReturnValue(mockedDispatch);
    useSelectorSpy.mockReturnValue({
      data: {
        30821604: {
          by: 'cute_boi',
          id: 30821604,
          kids: [30821923],
          parent: 30820894,
          text: 'comment text',
          time: 1648398990,
          type: 'comment',
        },
      },

      error: null,
      isFetching: false,
    });
    jest.setTimeout(10000);
  });

  const props = {
    navigation: navigation,
  };

  const changeState = {
    ...initialState,
  };
  it('should match snapshot', () => {
    const wrapper = render(
      <Provider store={createMockStore(changeState)}>
        <Details {...props} />
      </Provider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('Should contain CLick More Text', () => {
    const {queryByTestId} = render(
      <Provider store={createMockStore(changeState)}>
        <Details {...props} />
      </Provider>,
    );
    expect(queryByTestId('detail-feed-text')).toBeNull();
  });
});
