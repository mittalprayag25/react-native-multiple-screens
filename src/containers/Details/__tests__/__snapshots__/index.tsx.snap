// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details with isFetching data should match snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ff6600",
          "height": 50,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Hacker News
      </Text>
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        new | Past | comments | ask | show | jobs
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      />
      <Text>
        Comments
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
            "justifyContent": "space-around",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`Details with success data should match snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ff6600",
          "height": 50,
          "justifyContent": "center",
          "paddingLeft": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Hacker News
      </Text>
      <Text
        style={
          Object {
            "color": "black",
          }
        }
      >
        new | Past | comments | ask | show | jobs
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      />
      <Text>
        Comments
      </Text>
      <Text
        testID="detail-feed-text"
      >
        Fuchsia Workstation
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
            "justifyContent": "space-around",
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
